src/test/java/be/syntra/devshop/DevshopFront/services/CartServiceImplTest.java

```diff
@@ -75,71 +92,93 @@ public void getNewCartTest() {
 
         // then
-        assertEquals(getCart.getProducts().size(), 1);
+        assertEquals(getCart.getCartProductDtoList().size(), 1);
         assertEquals(getCart.isFinalizedCart(), false);
         assertEquals(getCart.isPaidCart(), false);
     }
 
     @Test
     void canAddOneToProductInCartTest() {
         // given
-        cartService.addToCart(ProductUtils.getDummyNonArchivedProduct());
+        final ProductDto dummyProductDto = getOtherDummyProductDto();
+        final Product dummyProduct = getDummyNonArchivedProduct();
+        when(productService.findById(dummyProductDto.getId())).thenReturn(dummyProduct);
+        when(productMapper.convertToProductDto(dummyProduct)).thenReturn(dummyProductDto);
+        cartService.addToCart(dummyProductDto.getId());
 
         // when
-        cartService.addOneToProductInCart(1L);
+        cartService.addOneToProductInCart(dummyProduct.getId());
 
         // then
-        assertEquals(currentCart.getProducts().get(0).getTotalInCart(), 3);
+        //assertEquals(currentCart.getProductDtos().get(0).getTotalInCart(), 3);
+        assertEquals(currentCart.getCartProductDtoList().get(0).getCount(), 2);
     }
 
     @Test
     void canRemoveOneFromProductInCartTest() {
         // given
-        cartService.addToCart(ProductUtils.getDummyNonArchivedProduct());
-        currentCart.getProducts().get(0).setTotalInCart(3);
+        final ProductDto dummyProductDto = getOtherDummyProductDto();
+        final Product dummyProduct = getDummyNonArchivedProduct();
+        when(productService.findById(dummyProductDto.getId())).thenReturn(dummyProduct);
+        when(productMapper.convertToProductDto(dummyProduct)).thenReturn(dummyProductDto);
+        cartService.addToCart(dummyProductDto.getId());
+        //currentCart.getProductDtos().get(0).setTotalInCart(3);
+        currentCart.getCartProductDtoList().get(0).setCount(3);
 
         // when
         cartService.removeOneFromProductInCart(1L);
 
         // then
-        assertEquals(currentCart.getProducts().get(0).getTotalInCart(), 2);
+        //assertEquals(currentCart.getProductDtos().get(0).getTotalInCart(), 2);
+        assertEquals(currentCart.getCartProductDtoList().get(0).getCount(), 2);
     }
 
     @Test
     void canRemoveProductFromCartTest() {
         // given
-        cartService.addToCart(ProductUtils.getDummyNonArchivedProduct());
+        final ProductDto dummyProductDto = getOtherDummyProductDto();
+        final Product dummyProduct = getDummyNonArchivedProduct();
+        when(productService.findById(dummyProductDto.getId())).thenReturn(dummyProduct);
+        when(productMapper.convertToProductDto(dummyProduct)).thenReturn(dummyProductDto);
+        cartService.addToCart(dummyProductDto.getId());
 
         // when
         cartService.removeProductFromCart(1L);
 
         // then
-        assertEquals(currentCart.getProducts().size(), 0);
+        assertEquals(currentCart.getCartProductDtoList().size(), 0);
```

Comment:
sidenote: assertEquals() expects the first parameter to be the 'expected', and the second as the 'actual' (or to be asserted) argument. This is why I prefer the assertThat() notation

Reply:
haha, the ordering of my args dictates that a should be using assertThat() so my habit is wrong now grrr